src/controllers/userProfileController.js

```diff
@@ -236,7 +236,7 @@ const userProfileController = function (UserProfile) {
     const isRequestorAuthorized = !!(
       canRequestorUpdateUser(req.body.requestor.requestorId, userid) && (
         await hasPermission(req.body.requestor, 'putUserProfile')
-        || req.body.requestor.requestorId === userid
+        || (await hasPermission(req.body.requestor, "modifyBadgeAmount")) || req.body.requestor.requestorId === userid
```

Comment:
Need to limit 'putUserProfile' and 'modifyBadgeAmount' to only modify their related data.